### PR TITLE
[ci] Build with arcade 

### PR DESCRIPTION
**Pull request by rmarinho**
### Description of Change

Adds a new way to build on the internal repo with arcade

This pull request includes several updates to configurations, build scripts, and dependency versions. The most important changes focus on updating dependency versions, modifying build scripts, and updating package sources.

### Dependency Updates:
* Updated `microsoft.dotnet.xharness.cli` version from `9.0.0-prerelease.24420.3` to `8.0.0-prerelease.24452.3` in `.config/dotnet-tools.json`.
* Updated dependency versions for `Microsoft.DotNet.XHarness.*` and `Microsoft.DotNet.Build.Tasks.Feed` in `eng/Version.Details.xml` and `eng/Versions.props`. [[1]](diffhunk://#diff-fb62e94a1d6f29f863e3d0a22aa38269f6cd1d7f03b109dc06e2cbf2548b86d3L3-R23) [[2]](diffhunk://#diff-1ea18ff65faa2ae6fed570b83747086d0317f5e4bc325064f6c14319a9c4ff67L108-R117)

### Build Script Modifications:
* Simplified `build.cmd` to handle arguments and invoke `build.ps1` script.
* Removed old `build.ps1` script.
* Updated `build.sh` to resolve symlinks and call `eng/build.sh`.

### Configuration Changes:
* Commented out `SymbolPackageFormat` in `Directory.Build.props`.
* Changed `DotNetDirectory` to use `$(RepoRoot)/.dotnet/` in `Directory.Build.props`.
* Added new package sources and removed old ones in `NuGet.config`.

### Other Changes:
* Added new properties and warnings to `Directory.Build.targets`.
* Added new build and signing configurations in `eng/Build.props`, `eng/Signing.props`, and `eng/Tools.props`. [[1]](diffhunk://#diff-366b25472963c6b33188d84139f6c8997298c5adc857a2624ab0d26a2d87baf0R1-R7) [[2]](diffhunk://#diff-a923a7746e08ef5df4bfd49f340d9f93055d66ba792770a7e1c227b3eaf3d4efL3-L43) [[3]](diffhunk://#diff-008293a45f05d88c0dba977ec62198b6f0f011e02e44619fa09d9f58e82cded7R1-R13)

These changes aim to streamline the build process, update dependencies to newer versions, and improve configuration management.

### CODE REVIEW

File: Directory.Build.targets [hunk 1]
```diff
@@ -19,10 +19,16 @@
...
   <PropertyGroup>
-     <!-- Allows for MAUI Xaml Hot Reload Samples to run without checks  -->
+    <!-- Allows for MAUI Xaml Hot Reload Samples to run without checks  -->
     <IgnoreMauiXamlHotReloadCompatibilityCheck>True</IgnoreMauiXamlHotReloadCompatibilityCheck>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <!-- Microsoft.Build.Msix.props" cannot be imported again -->
+    <!-- Found version-specific or distribution-specific runtime identifier(s)-->
+    <NoWarn>$(NoWarn);MSB4011;NETSDK1206</NoWarn>
```
Comment (rmarinho) [inline Directory.Build.targets:29]: this doesn't t actually seem to work, but would be good to try to hide these warnings or fix them @mattleibow

File: NuGet.config [hunk 2]
```diff
@@ -4,50 +4,19 @@
...
+    <add key="dotnet-libraries" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-libraries/nuget/v3/index.json" />
+    <add key="dotnet-libraries-transport" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-libraries-transport/nuget/v3/index.json" />
+    <add key="vs-buildservices" value="https://pkgs.dev.azure.com/azure-public/vside/_packaging/vs-buildservices/nuget/v3/index.json" />
+    <add key="dotnet9" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet9/nuget/v3/index.json" />
+    <add key="dotnet9-transport" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet9-transport/nuget/v3/index.json" />
   </packageSources>
   <activePackageSource>
     <add key="All" value="(Aggregate source)" />
   </activePackageSource>
-  <disabledPackageSources>
```
Comment (rmarinho) [inline NuGet.config:18]: I need to add this back

File: Directory.Build.props [hunk 3]
```diff
@@ -191,9 +190,8 @@
   </PropertyGroup>
 
   <PropertyGroup>
-    <DotNetOutputPath>$(MSBuildThisFileDirectory)bin/</DotNetOutputPath>
-    <DotNetTempDirectory>$(DotNetOutputPath)temp/</DotNetTempDirectory>
-    <DotNetDirectory>$(DotNetOutputPath)dotnet/</DotNetDirectory>
+    <DotNetTempDirectory>$(RepoRoot)/temp/</DotNetTempDirectory>
```
Comment (mattleibow) [inline Directory.Build.props:193]: Where does `RepoRoot` come from and if I run a build in VS will it be a blank string making this evaluate to C:\temp?

File: global.json [hunk 4]
```diff
@@ -1,13 +1,13 @@
 {
   "tools": {
-    "dotnet": "9.0.100-preview.4.24267.66"
+    "dotnet": "8.0.108"
```
[the remark this answers is not in the document]
Comment (rmarinho) [reply global.json:3]: Arcade itself doesn't have away to install workloads from from the Version.props I wonder if we can make a PR to arcade with Dotnet.csproj and add a way to globally install the sdk and workloads specified or public

File: src/Workload/maui.vsmanproj [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+  <Import Project="$(SwixBuildTargets)"/>
+
+  <Target Name="SetXamarinDropVars" >
+    <Message Text="##vso[task.setvariable variable=XamarinDropName;isOutput=true]$(ManifestDropName)" Importance="high" />
+    <Message Text="##vso[task.setvariable variable=XamarinDropManifestName;isOutput=true]$(ManifestName).vsman" Importance="high" />
+    <Message Text="##vso[task.setvariable variable=XamarinDropManifestVersion;isOutput=true]$(ManifestBuildVersion)" Importance="high" />
+    <PropertyGroup>
+      <_DropVars>
+XamarinDropName=$(ManifestDropName)
```
Comment (rmarinho) [inline src/Workload/maui.vsmanproj:93]: @pjcollins does this look good with Xamarin naming ? can we rename this?

File: src/Graphics/src/Graphics/Graphics.csproj [hunk 6]
```diff
@@ -4,7 +4,6 @@
     <TargetFrameworks>netstandard2.1;netstandard2.0;$(_MauiDotNetTfm);$(MauiGraphicsPlatforms);$(_MauiPreviousDotNetTfm);$(MauiGraphicsPreviousPlatforms)</TargetFrameworks>
     <!-- Optional: Build symbol package (.snupkg) to distribute the PDB containing Source Link -->
     <IncludeSymbols>true</IncludeSymbols>
-    <SymbolPackageFormat>snupkg</SymbolPackageFormat>
```
Comment (rmarinho) [inline src/Graphics/src/Graphics/Graphics.csproj:7]: @mattleibow  It s is generating them , but I needed to remove it from here. 

But the are on the BlobArtifacts folder https://dev.azure.com/dnceng/7ea9116e-9fac-403d-b258-b31fcf1bb293/_apis/build/builds/2535536/artifacts?artifactName=BlobArtifacts&api-version=7.1&%24format=zip

File: src/Workload/maui.vsmanproj (hunk 5 above)
Comment (pjcollins) [reply src/Workload/maui.vsmanproj:93]: With the latest v4 of the MSI convert template this SetXamarinDropVars target is no longer used, so it should be safe to remove this entirely.

File: src/Workload/workloads.csproj [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <WixToolsetPath>$(PkgMicrosoft_Signed_Wix)\tools</WixToolsetPath>
+    <SwixPluginPath>$(PkgMicroBuild_Plugins_SwixBuild_Dotnet)</SwixPluginPath>
+    <SwixBuildTargets>$(SwixPluginPath)\build\MicroBuild.Plugins.SwixBuild.targets</SwixBuildTargets>
+    <SwixTargetsPath>$(PkgMicroBuild_Plugins_SwixBuild_Dotnet)\build\MicroBuild.Plugins.SwixBuild.targets</SwixTargetsPath>
+  </PropertyGroup>
+
+  <Import Project="$(WorkloadMsiGenProps)" />
```
Comment (pjcollins) [inline src/Workload/workloads.csproj:30]: If we are no longer going to be sharing this file across repos, we could also get rid of the separate vs-workload.props file which contains product/repo specific information. Those props/items could be moved in here instead.

File: Directory.Build.props (hunk 3 above)
Comment (rmarinho) [reply Directory.Build.props:193]: just tried our cake on vs and it works, this temp dir is just install the .net install script and workload packs before installing the iOS and android workloads
. 

Can you take these for a run and try the build.sh -pack command ?

Comment (rmarinho) [reply Directory.Build.props:193]: I got it to pack fine but I think I might be missing the NativeAssets thing ?

**Pull request closed without merging**